// @storybook/addon-ondevice-backgrounds/dist/index.js
"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var addons_1 = __importStar(require("@storybook/addons"));
var constants_1 = __importDefault(require("./constants"));
var container_1 = __importDefault(require("./container"));
exports.withBackgrounds = addons_1.makeDecorator({
    name: 'withBackgrounds',
    parameterName: 'backgrounds',
    skipIfNoParametersOrOptions: true,
    allowDeprecatedUsage: true,
    wrapper: function (getStory, context, _a) {
        var options = _a.options, parameters = _a.parameters;
        var data = parameters || options || [];
        var backgrounds = Array.isArray(data) ? data : Object.values(data);
        var background = 'transparent';
        if (backgrounds.length !== 0) {
            addons_1.default.getChannel().emit(constants_1.default.SET, backgrounds);
            var defaultOrFirst = backgrounds.find(function (x) { return x.default; }) || backgrounds[0];
            if (defaultOrFirst) {
                background = defaultOrFirst.value;
            }
        }
        return (React.createElement(container_1.default, { initialBackground: background, channel: addons_1.default.getChannel() }, getStory(context)));
    },
});
