// @storybook/addon-ondevice-backgrounds/dist/Swatch.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var prop_types_1 = __importDefault(require("prop-types"));
var react_native_1 = require("react-native");
var Swatch = function (_a) {
    var name = _a.name, value = _a.value, setBackground = _a.setBackground;
    return (react_1.default.createElement(react_native_1.TouchableOpacity, { style: {
            borderRadius: 4,
            borderWidth: 1,
            borderColor: 'rgba(0,0,0,0.2)',
            marginTop: 10,
            marginBottom: 20,
            marginHorizontal: 10,
        }, onPress: function () { return setBackground(value); } },
        react_1.default.createElement(react_native_1.View, { style: { flex: 1, backgroundColor: value, height: 40 } }),
        react_1.default.createElement(react_native_1.View, { style: { padding: 4, flexDirection: 'row', justifyContent: 'space-between' } },
            react_1.default.createElement(react_native_1.Text, null,
                name,
                ":"),
            react_1.default.createElement(react_native_1.Text, null, value))));
};
Swatch.propTypes = {
    name: prop_types_1.default.string.isRequired,
    value: prop_types_1.default.string.isRequired,
    setBackground: prop_types_1.default.func.isRequired,
};
exports.default = Swatch;
